independent-projects/tools/devtools-testing/src/main/java/io/quarkus/devtools/testing/SnapshotTesting.java

```diff
@@ -0,0 +1,244 @@
+        final boolean updateSnapshot = shouldUpdateSnapshot(snapshotName);
+
+        if (updateSnapshot && Files.isRegularFile(snapshotFile)) {
+            deleteExistingSnapshots(snapshotName, snapshotFile);
+        }
+
+        if (!Files.isRegularFile(snapshotFile) || updateSnapshot) {
+            Files.createDirectories(snapshotFile.getParent());
+            Files.write(snapshotFile, String.join("\n", tree).getBytes(StandardCharsets.UTF_8));
+        }
+
+        assertThat(snapshotFile).isRegularFile();
+
+        final List<String> content = Arrays.stream(getContent(snapshotFile).split("\\v"))
+                .filter(s -> !s.isEmpty())
+                .collect(toList());
+
+        return assertThat(tree).containsExactlyInAnyOrderElementsOf(content);
+    }
+
+    public static byte[] getBinaryContent(Path file) {
+        try {
+            return Files.readAllBytes(file);
+        } catch (IOException e) {
+            throw new UncheckedIOException("Unable to read " + file.toString(), e);
+        }
+    }
+
+    public static String getContent(Path file) {
+        try {
+            return new String(Files.readAllBytes(file), StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            throw new UncheckedIOException("Unable to read " + file.toString(), e);
+        }
+    }
+
+    public static void deleteTestDirectory(final File file) throws IOException {
+        FileUtils.deleteDirectory(file);
+
+        Assertions.assertFalse(
+                Files.exists(file.toPath()), "Directory still exists");
+    }
+
+    /**
+     * To use with {@link AbstractPathAssert} in order to check the file content contains a specific string.
+     *
+     * @param s the string which should be in the file content
+     * @return a {@link Consumer<Path>} to use with {@link AbstractPathAssert#satisfies(Consumer)}
+     */
+    public static Consumer<Path> checkContains(String s) {
+        return (p) -> assertThat(getContent(p)).contains(s);
+    }
+
+    public static Consumer<Path> checkMatches(String regex) {
+        return (p) -> assertThat(getContent(p)).matches(regex);
+    }
+
+    private static boolean shouldUpdateSnapshot(String identifier) {
+        return getUpdateSnapshotsProp().filter(u -> u.isEmpty() || "true".equalsIgnoreCase(u) || u.contains(identifier))
+                .isPresent();
+    }
+
+    private static boolean isUTF8File(final Path file) {
+        try {
+            final byte[] inputBytes = Files.readAllBytes(file);
+            final String converted = new String(inputBytes, StandardCharsets.UTF_8);
+            final byte[] outputBytes = converted.getBytes(StandardCharsets.UTF_8);
+            return Arrays.equals(inputBytes, outputBytes);
+        } catch (IOException e) {
+            return false;
+        }
+    }
+
+    private static void deleteExistingSnapshots(String name, Path snapshots) {
+        System.out.println("\n>>>>>> DELETING EXISTING TEST SNAPSHOTS FOR:\n>>>>>> " + name + "\n");
```

Comment:
Is this System.out necessary or can it use a logger instead?

Reply:
It's used only in the tests, goal is to make it really obvious, log are not always displayed..

Reply:
You only see it when using `-Dsnap`